### DISCUSSION TITLE
statically compiling program usign HighFive library

### DISCUSSION THREAD

**Discussion by punims** [BlueBrain/HighFive#612] [Questions] [answered]
Hello, 
we've been trying to compile a program completely statically while using your library. 
We noticed that HighFive uses hdf5 which in turn uses Szip and Szip apparently uses a shared library (AEC) so the program cannot compile. Are there any suggestions for us on how to proceed?

**Comment by pramodk** [comment, marked as answer]
@punims : for the szip, you can see that there are options to build it statically:

```console
$ ./configure --help
...
  --enable-shared[=PKGS]  build shared libraries [default=yes]
  --enable-static[=PKGS]  build static libraries [default=yes]
...
```

So `--enable-shared=no  --enable-static=yes` should do the job.

By the way, szip is only needed if you want to enable szip-based compression. If that is not required, with cmake build system of hdf5, one can use options like:

```console
-DHDF5_ENABLE_Z_LIB_SUPPORT:BOOL=ON -DHDF5_ENABLE_SZIP_SUPPORT:BOOL=OFF -DBUILD_SHARED_LIBS:BOOL=OFF -DONLY_SHARED_LIBS:BOOL=OFF
```

Hope that helps!
